public/app/plugins/datasource/elasticsearch/elastic_response.ts

```diff
@@ -516,7 +535,40 @@ export class ElasticResponse {
       }
     }
 
-    return { data: seriesList };
+    return { data: seriesList, key: this.targets[0]?.refId };
+  };
+
+  processPPLResponseToSeries = () => {
+    const target = this.targets[0];
+    const response = this.response;
+    // We check if any valid date type is contained in the response
+    const timeFieldIndex = _.findIndex(
+      response.schema,
+      (field: { type: string }) =>
+        field.type === 'timestamp' || field.type === 'datetime' || field.type === 'date' || field.type === 'time'
+    );
+    const valueIndex = timeFieldIndex === 0 ? 1 : 0;
+
+    //time series response should include a value field and timestamp
+    if (timeFieldIndex === -1 || response.datarows[0].length !== 2 || isNaN(response.datarows[0][valueIndex])) {
+      throw this.getInvalidPPLQuery(this.response);
+    }
+
+    const datapoints = _.map(response.datarows, datarow => {
+      const newDatarow = _.clone(datarow);
+      const [timestamp] = newDatarow.splice(timeFieldIndex, 1);
```

Comment:
Could you find the way to hide the timeFieldIndex from usage. e.g. 
```
class DataRow {
  Timestamp getTimeStamp();
}
```

Reply:
Could you clarify what you mean by this?